src/Symfony/Component/Form/FormRegistry.php

```diff
@@ -97,13 +98,34 @@ public function getType($name)
      */
     private function resolveAndAddType(FormTypeInterface $type)
     {
+        static $checkedTypes = array();
+
+        $searchKey = array_search($type->getName(), $checkedTypes);
+        $checkedTypes[] = $type->getName();
```

Comment:
would putting the name as key work instead?

Reply:
If the name is the key, then you won't get the proper dependency path when throwing an error,
E.G if you have the dependencies 'foo -> bar -> foo' then you can only have the values `foo > bar` in the array that is displayed in the message, where we want to show `foo > bar > foo` (So we want to be able to have duplicate values in the array)

Reply:
Is it possible to have duplicate value and *not* have a circular ref? If not, then shouldn't this be just the terminal condition? (ie the last `> foo` comes from the current call, not from the checkedTypes stack?)

Reply:
You're right, any duplicate value would immediately be the cause of the circular ref. I have updated this